### ISSUE TITLE
Stack traces in sentry dashboard contain unsymbolicated lines when using --split-debug-info

### ISSUE THREAD

**Issue by acoutts** [getsentry/sentry-dart#1824]
### Platform

Flutter Mobile

### Obfuscation

Disabled

### Debug Info

Enabled

### Doctor

[!] Flutter (Channel stable, 3.16.5, on macOS 14.2.1 23C71 darwin-arm64, locale en-US)
    • Flutter version 3.16.5 on channel stable at /Users/andrewcoutts/fvm/versions/3.16.5
    ! Warning: `dart` on your path resolves to /Users/andrewcoutts/Projects/flutter/bin/cache/dart-sdk/bin/dart, which is not inside your current Flutter SDK checkout at /Users/andrewcoutts/fvm/versions/3.16.5. Consider adding /Users/andrewcoutts/fvm/versions/3.16.5/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 78666c8dc5 (4 weeks ago), 2023-12-19 16:14:14 -0800
    • Engine revision 3f3e560236
    • Dart version 3.2.3
    • DevTools version 2.28.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/andrewcoutts/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.80.0

[✓] VS Code (version 1.85.0-insider)
    • VS Code at /Applications/Visual Studio Code - Insiders.app/Contents
    • Flutter extension version 3.79.20231130

[✓] Connected device (4 available)
    • xff0c-13p (mobile)         • 00008110-000E0D090199401E            • ios            • iOS 17.2.1 21C66
    • iPhone 15 Pro Max (mobile) • A6742E39-0F9D-442B-B225-BEECD609A2C2 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 14.2.1 23C71 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 120.0.6099.216

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

### Version

7.14.0

### Steps to Reproduce

1. I create a build with `split-debug-info`
2. trigger an error in the app

### Expected Result

The stack traces should look the same with split-debug-info enabled or disabled.

### Actual Result

The errors contain the unsymbolicated header:

```
<unknown>
#0 *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** (unparsed)
In App
<unknown>
#1 pid: 16778, tid: 6133592064, name io.flutter.1.ui (unparsed)
In App
<unknown>
#2 os: ios arch: arm64 comp: no sim: no (unparsed)
In App
<unknown>
#3 build_id: '04db4a650e0e38f39b6da68bb42bc448' (unparsed)
In App
<unknown>
#4 isolate_dso_base: 10b778000, vm_dso_base: 10b778000 (unparsed)
In App
<unknown>
#5 isolate_instructions: 10b786340, vm_instructions: 10b77c000 (unparsed)
```

It causes the sentry event source to be:
```
_Exception unparsed in #0 *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
```
instead of the real exception, which makes it difficult to find errors in the console.

Inside the stack trace, a few lines down is the actual symbolicated code:
<img width="798" alt="Screenshot 2024-01-17 at 1 33 17 PM" src="https://github.com/getsentry/sentry-dart/assets/9597207/b870c4fa-55cd-488b-ad11-20c6a3db04ce">

If I turn off split-debug-symbols then the problem doesn't happen but then I cannot see the source code in the dashboard:
<img width="797" alt="Screenshot 2024-01-17 at 1 34 07 PM" src="https://github.com/getsentry/sentry-dart/assets/9597207/211cfb36-3510-41da-b18a-417f7426cdc0">

### Are you willing to submit a PR?

Yes

**Comment by buenaflor** [comment]
Hi, does this affect both iOS and Android?

**Comment by acoutts** [comment]
Yup it's on android as well.
<img width="805" alt="Screenshot 2024-01-18 at 10 42 34 AM" src="https://github.com/getsentry/sentry-dart/assets/9597207/63912056-9ebf-44ff-8177-70a2381cd610">

**Comment by buenaflor** [comment]
Do you have a sample app where we can reproduce this? 

I created a new flutter project, installed sentry

- `flutter build apk --split-debug-info=your/path --obfuscate`
- `dart run sentry_dart_plugin`
- trigger error

works for me

**Comment by acoutts** [comment]
what if you don't obfuscate? I am not obfuscating right now could that be it?

**Comment by buenaflor** [comment]
Still works without --obfuscate, if you have time could you prepare a reproducible sample or steps so we can take a closer look?

**Comment by acoutts** [comment]
Can you post a picture of what the logged events look like in your case? I just want to compare it and make sure we're talking about the same thing before I go and try to create a new app with a new sentry instance which will take quite some time.

**Comment by buenaflor** [comment]
<img width="1124" alt="image" src="https://github.com/getsentry/sentry-dart/assets/23364143/e0eaa8c4-5206-475e-96ad-ef8e99edff46">

done with `flutter build apk --split-debug-info=path` same result with `flutter build apk --split-debug-info=path --obfuscate`

although it seems there's some problem with the source context for inApp frames

**Comment by acoutts** [comment]
I see - i can't nail down why it only happens on some of my events. But, I think I have a solution! Playing with the stack trace rules I was able to come to a solution:

```
app:no                   +app
stack.function:*SafeResponseEx* ^-app -app ^-group -group
stack.function:*ChopperClient* ^-app -app ^-group -group
stack.function:*FutureHandlerProviderElementMixin* -app -group
```

On the garbage events I can see the unparsed frames being considered for the grouping/in-app rule:
<img width="944" alt="Screenshot 2024-01-19 at 9 31 42 AM" src="https://github.com/getsentry/sentry-dart/assets/9597207/d4b3fb7f-b45a-4292-b0f1-43f20db9e9d5">

I was unable to filter out the lines using the matcher `unparsed` but I was able to do it by telling it to cut the stack trace above/below certain points to exclude the unparsed frames. This seems to have cleaned things up for me.

I'm not sure what the parsing issue is but it seems related to the split-debug flag because when I remove that there are no issues, but of course then source context is missing. Hopefully it will be fixed by https://github.com/getsentry/sentry-dart/issues/1314.
